### PR TITLE
chore(weave): change raise to print for netrc existing with api key

### PR DESCRIPTION
**Pull request by jwlee64**
## Description
Changes the reaction to netrc and apikey existing to a print and not a raise when not in prod

## Testing

How was this PR tested?

### CODE REVIEW

File: weave/legacy/weave/environment.py [hunk 1]
```diff
@@ -251,9 +251,7 @@ def weave_wandb_api_key() -> typing.Optional[str]:
     env_api_key = _wandb_api_key_via_env()
     netrc_api_key = _wandb_api_key_via_netrc()
     if env_api_key and netrc_api_key:
-        raise errors.WeaveConfigurationError(
-            "WANDB_API_KEY should not be set in both ~/.netrc and the environment."
-        )
+        print("WANDB_API_KEY should not be set in both ~/.netrc and the environment.")
```
Comment (jamie-rasmussen) [inline weave/legacy/weave/environment.py:254]: * If the contents are identical I think we could be silent.
* I'd change the message to explain a bit better. Maybe something like:
"There are different credentials in the netrc file and the environment. Using the environment value." 
* Not worth fixing but just noting that NETRC file might not be located at ~/.netrc (Windows, I think we might support the NETRC env var for alternate path, etc.)

Comment (jwlee64) [reply weave/legacy/weave/environment.py]: makes sense
_wandb_api_key_via_netrc() -> i believe checks the windows location as well

Comment (jamie-rasmussen) [reply weave/legacy/weave/environment.py]: Right, just meant that on windows this message would have an incorrect path (`~/.netrc` vs. `~/_netrc`)

File: weave/legacy/weave/environment.py [hunk 2]
```diff
@@ -250,9 +250,9 @@ def _wandb_api_key_via_netrc_file(filepath: str) -> typing.Optional[str]:
 def weave_wandb_api_key() -> typing.Optional[str]:
     env_api_key = _wandb_api_key_via_env()
     netrc_api_key = _wandb_api_key_via_netrc()
-    if env_api_key and netrc_api_key:
-        raise errors.WeaveConfigurationError(
-            "WANDB_API_KEY should not be set in both ~/.netrc and the environment."
+    if env_api_key and netrc_api_key and env_api_key != netrc_api_key:
```
Comment (tssweeney) [inline weave/legacy/weave/environment.py:253]: i would still raise in production

File: weave/legacy/weave/environment.py [hunk 3]
```diff
@@ -250,10 +250,14 @@ def _wandb_api_key_via_netrc_file(filepath: str) -> typing.Optional[str]:
 def weave_wandb_api_key() -> typing.Optional[str]:
     env_api_key = _wandb_api_key_via_env()
     netrc_api_key = _wandb_api_key_via_netrc()
-    if env_api_key and netrc_api_key:
+    if env_api_key and netrc_api_key and wandb_production():
```
Comment (tssweeney) [inline weave/legacy/weave/environment.py:253]: nit:

```
    if env_api_key and netrc_api_key:
        if wandb_production():
          raise errors.WeaveConfigurationError(
              "WANDB_API_KEY should not be set in both ~/.netrc and the environment."
          )
        elif env_api_key != netrc_api_key:
          print(
              "There are different credentials in the netrc file and the environment. Using the environment value."
          )
```

**Pull request merged**